File: backend/src/main/java/com/boatsched/entity/SystemConfig.java
package com.boatsched.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "system_config")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SystemConfig {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT")
    private String noticeTitle;

    @Column(columnDefinition = "TEXT")
    private String noticeContent;

    @Column(columnDefinition = "TEXT")
    private String popupContent;

    @Column(nullable = false)
    private Boolean isDangerMode = false;
}

File: frontend/lib/models/system_config.dart
class SystemConfig {
  final String? noticeTitle;
  final String? noticeContent;
  final String? popupContent;
  final bool isDangerMode;

  SystemConfig({
    this.noticeTitle,
    this.noticeContent,
    this.popupContent,
    required this.isDangerMode,
  });

  factory SystemConfig.fromJson(Map<String, dynamic> json) {
    return SystemConfig(
      noticeTitle: json['noticeTitle'],
      noticeContent: json['noticeContent'],
      popupContent: json['popupContent'],
      isDangerMode: json['isDangerMode'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'noticeTitle': noticeTitle,
      'noticeContent': noticeContent,
      'popupContent': popupContent,
      'isDangerMode': isDangerMode,
    };
  }
}
